client.tf

```diff
@@ -1,11 +1,11 @@
 resource "aws_instance" "client" {
-  count                       = var.clients
+  count                       = var.client_count  * length(var.availability_zone)
   ami                         = coalesce(var.client_ami, data.aws_ami.ami.image_id)
   instance_type               = var.client_instance_type
   key_name                    = aws_key_pair.ssh.key_name
-  subnet_id                   = var.subnet_id
-  vpc_security_group_ids      = concat([aws_security_group.node_sec_group.id], var.security_groups_client)
-  associate_public_ip_address = var.associate_public_ip_addr
+  subnet_id                   = try(lookup(local.merged_subnets["client"], var.availability_zone[count.index % length(var.availability_zone)]), "")
```

Comment:
Not quite sure I understand what's happening here.

We index of the client instance and mod it by the number of AZs to come up with the name of an AZ, and then look that up in the map of merged subnets to figure out what subnet id to use for that az?

Reply:
```  subnet_id                   = try(lookup(local.merged_subnets["client"], var.availability_zone[count.index % length(var.availability_zone)]), "")
```

So I'll go thru each function:

* try: this is a try-catch. If the contained functions throw an error it replaces the erroneous output with "". This results in us using the default subnet on the default vpc in practice. If there is no default vpc/subnet, then it errors out again and ends processing.

* lookup: we assume that the _count provided is intended to be created in each AZ rather than spread between them (so broker_count=3 means 3 per az). Because we have a list of AZ we can do a modulo on the count with the length of the AZ list to pick the correct AZ from the list of AZ. We then use that to do a lookup on the merged_subnets for client, which allows us to correctly pick the subnets for that AZ.